src/components/Hoverable/index.js

```diff
@@ -1,189 +1,198 @@
 import _ from 'underscore';
-import React, {Component} from 'react';
+import React, {useEffect, useCallback, useState, useRef, useMemo, useImperativeHandle} from 'react';
 import {DeviceEventEmitter} from 'react-native';
 import {propTypes, defaultProps} from './hoverablePropTypes';
 import * as DeviceCapabilities from '../../libs/DeviceCapabilities';
 import CONST from '../../CONST';
 
+function mapChildren(children, callbackParam) {
```

Comment:
I think about adding comments for `mapChildren` and `assignRef` to clarify why we have this logic.

Reply:
Thanks for the idea, done ✅

Reply:
Thank you, looks good!

Reply:
@kacper-mikolajczak, do we need to extract this logic outside of the component? Usually, if logic is extracted should be moved to utils or helpers dir

Reply:
Those helpers are not really that universal to move them into utils - I did not want to bloat the component body no more, thus extracted them away. This might help to reduce the cognitive load when studying the component, wdyt?

Is there a convention we use for such local helpers? Shall we create a separate file under component's directory called `utils`/`helpers` or just leave them inside the component's file (current approach)?

Thanks for pointing this out!

Reply:
I agree that local functions simplify readability. For me, it can stay as it is. But `assignRef` seems to be a universal function. It could be moved to, e.g. react utils or helpers. @Beamanator, any thoughts?